### PR TITLE
translate csv data to map to current model and its relevant attributes

### PR DESCRIPTION
**Pull request by sooyounga**
based on provided small data set of 20230412, the columns are mapped to these attributes in our model:

- genome_annotation_label --> `synonym`
- gene_identifier_prefix --> `id` prefix (stylized)
- gene_local_unique_identifier --> `id` suffix
- symbol --> `symbol`
- name --> `name`
- gene_biotype --> NOT mapped

translator.py is first draft of how we can take such data in csv format and load it into yaml data format to load into our model class

### CODE REVIEW

**Comment by satra** [comment]
let's add a slot in our model for `biotype` called `molecular type` and lets' make the range of that slot be an enum with two values: `protein-coding` and `noncoding`.

also once the other csv mapping is done, let's use that to add values to the  `in_taxon` and `referenced in` slots.

for the other csvs, we want to collate and map that information into `GenomeAnnotation`, whose `id` would then by used for the value of `referenced in`.

**Comment by satra** [comment]
is this rebased on current master? seems like a lot of unrelated files have been updated.

File: data-examples/data.yaml [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+  referenced_in: Ensembl-10090-108
+  symbol: Pdyn
+- id: ENSEMBL:ENSMUSG00000027400
+  molecular_type: protein_coding
+  name: prodynorphin [Source:MGI Symbol%3BAcc:MGI:97535]
+  referenced_in: Ensembl-10090-109
+  symbol: Pdyn
+genome annotations:
+- content_url:
+  - https://ftp.ensembl.org/pub/release-102/gff3/mus_musculus/Mus_musculus.GRCm38.102.gff3.gz
```
Comment (satra) [inline data-examples/data.yaml:54]: don't see the digest info in this record

File: linkml-schema/kbmodel.yaml [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+      - version
+      - digest
+      - content_url
+
+  genome assembly:
+    description: >-
+      Genome assembly to contain version and label information
+    attributes:
+      id:
+      taxon:
```
Comment (satra) [inline linkml-schema/kbmodel.yaml:46]: perhaps use the same mixin used by gene, such that `in_taxon` is a common reference?

Comment (sooyounga) [reply linkml-schema/kbmodel.yaml:49]: i tried to add mixins using `thing with taxon` as well as using `named thing` and `biological entity`, as well as directly quoting the python class syntax as seen in tutorial, but all raise the same error of `genome_assemblies -> 0 -> in_taxon
  extra fields not permitted (type=value_error.extra)` and i also tried directly modifying python classes generated to allow extra fields by `extra = Extra.allow` or ignore or direct strings of those enums, but they raise the same error again. so currently it is still just a separate class of separate fields. I'll continue looking into a possible way

**Pull request merged**